src/auto-evo/simulation/PopulationSimulation.cs

```diff
@@ -130,23 +133,115 @@ private static void SimulatePatchStep(RunResults populations, Patch patch, List<
+                totalPredationScore += GetPredationScore(currentMicrobeSpecies);
+            }
+
+            // Avoid division by 0
+            totalPhotosynthesisScore = Math.Max(1, totalPhotosynthesisScore);
+            totalChemosynthesisScore = Math.Max(1, totalChemosynthesisScore);
+            totalPredationScore = Math.Max(1, totalPredationScore);
+
+            // Calculate the share of environmental energy captured by each species
+            var energyAvailableForPredation = 0f;
+
+            foreach (MicrobeSpecies currentMicrobeSpecies in species)
             {
-                int currentPopulation = populations.GetPopulationInPatch(currentSpecies, patch);
-                int populationChange = random.Next(
-                    -Constants.AUTO_EVO_RANDOM_POPULATION_CHANGE, Constants.AUTO_EVO_RANDOM_POPULATION_CHANGE + 1);
+                var currentSpeciesEnergy = 0f;
+
+                currentSpeciesEnergy += sunlightInPatch
+                    * GetPhotosynthesisScore(currentMicrobeSpecies) / totalPhotosynthesisScore;
+                currentSpeciesEnergy += hydrogenSulfideInPatch
+                    * GetChemosynthesisScore(currentMicrobeSpecies) / totalChemosynthesisScore;
+
+                energyAvailableForPredation += 0.5f * currentSpeciesEnergy;
+                speciesEnergies.Add(currentMicrobeSpecies, currentSpeciesEnergy);
+            }
+
+            // Calculate the share of predation done by each species
+            // Then update populations
+            foreach (MicrobeSpecies currentMicrobeSpecies in species)
+            {
+                speciesEnergies[currentMicrobeSpecies] += energyAvailableForPredation
+                    * GetPredationScore(currentMicrobeSpecies) / totalPredationScore;
+
+                var newPopulation = (int)(speciesEnergies[currentMicrobeSpecies]
+                    / Math.Pow(currentMicrobeSpecies.Organelles.Count(), 1.3f));
 
                 if (lowSpecies)
                 {
-                    populationChange += Constants.AUTO_EVO_LOW_SPECIES_BOOST;
+                    newPopulation += Constants.AUTO_EVO_LOW_SPECIES_BOOST;
                 }
-                else if (highSpecies)
+
+                if (highSpecies)
                 {
-                    populationChange -= Constants.AUTO_EVO_HIGH_SPECIES_PENALTY;
+                    newPopulation -= Constants.AUTO_EVO_HIGH_SPECIES_PENALTY;
                 }
 
-                populations.AddPopulationResultForSpecies(currentSpecies, patch, currentPopulation + populationChange);
+                populations.AddPopulationResultForSpecies(currentMicrobeSpecies, patch, newPopulation);
             }
         }
+
+        private static float GetPhotosynthesisScore(MicrobeSpecies species)
+        {
+            var photosynthesisScore = 0f;
+            foreach (var organelle in species.Organelles)
+            {
+                if (organelle.Definition.InternalName == "chloroplast")
```

Comment:
This is not extensible, instead check if the organelle has a photosynthesis process, even that should probably be checked by finding a process with sunlight as the only input and making glucose.